### PR TITLE
feat: LoRA voice adaptation for VITS models

### PR DESCRIPTION
**Pull request by JarbasAl**
## LoRA Voice Adaptation for phoonnx

Add Low-Rank Adaptation (LoRA) support to phoonnx_train, enabling fast voice adaptation with less training data (5–15 min audio) by freezing the base model and only training lightweight adapter layers. Pronunciation is preserved from the base model since the TextEncoder is frozen.

### What's new

**Core LoRA infrastructure:**
- `phoonnx_train/vits/lora.py` — `LoRALinear`, `LoRAConv1d`, `LoRAConvTranspose1d` modules with `.merge()` for weight folding
- `phoonnx_train/vits/lora_config.py` — `LoRAConfig` dataclass with 3 presets:
  - `generator-only`: rank=4, targets `dec` only (~300KB adapter)
  - `full-acoustic`: rank=8, targets `dec,enc_q,flow,dp` (~1MB, default)
  - `aggressive`: rank=16, targets all including `enc_p` (~2.5MB)
- `phoonnx_train/vits/apply_lora.py` — `apply_lora()`, `merge_lora()`, `get_lora_state_dict()`, `load_lora_adapter()`, `count_parameters()` + automatic weight_norm stripping

**CLI integration:**
- `phoonnx_train/train.py` — new `--lora-scope`, `--lora-rank`, `--lora-alpha`, `--lora-target-modules`, `--lora-dropout` args; auto-saves adapter after training; auto-adjusts learning rate for LoRA
- `phoonnx_train/merge_lora.py` — CLI to merge LoRA adapter into base checkpoint and export to ONNX

**Inference-side:**
- `phoonnx/config.py` — `lora_base_model`, `lora_rank`, `lora_alpha`, `lora_scope`, `lora_target_modules` fields in `VoiceConfig`
- `phoonnx/voice.py` — `TTSVoice.load_with_lora()` for inference with merged LoRA
- `phoonnx/lora_runtime.py` — ONNX graph surgery for runtime LoRA injection

**Tests:** 21 new unit tests covering:
- Config validation (presets, invalid inputs)
- LoRA init identity (zero B → no change at init)
- Merge round-trip (apply → merge → verify same as manual)
- Freeze check (only LoRA params trainable)
- Adapter load/save
- All existing 121 tests still pass

### Usage

**Training with LoRA:**
```bash
python phoonnx_train/train.py \
  --dataset-dir /data/preprocessed \
  --resume-from-checkpoint /data/base.ckpt \
  --lora-scope full-acoustic \
  --max-epochs 200 --batch-size 16
```

**Merging for ONNX export:**
```bash
python phoonnx_train/merge_lora.py \
  /data/base.ckpt \
  --lora-adapter /data/lora_adapter/lora_adapter.pt \
  --config /data/config.json \
  --lora-scope full-acoustic \
  --output-dir /data/output/
```

### Wiki

Design plan documented at `knowledge/wiki/concepts/lora-phoonnx.md`

### CODE REVIEW

**Comment by JarbasAl** [comment]
> 🤖 Auto-generated by Claude Fable 5 (claude-fable-5) via Claude Code — NOT human-reviewed.

Stale by ~370 dev commits: the conflicts against current dev are substantive (phoonnx_train/config/voice surfaces all moved) and the green CI predates most of the current codebase, so it no longer evidences compatibility. The implementation (LoRA presets, weight-folding merge) is worth mining as reference when this is redone. Issue #136 remains open to track LoRA support against current dev; the branch is untouched.
